libvisual-plugins/plugins/input/alsa/input_alsa.c

```diff
@@ -167,14 +168,14 @@ int inp_alsa_init (VisPluginData *plugin)
 
        /* Setup a large buffer */
 
-	tmp = 1000000;
+	tmp = 1000 * 1000 / inp_alsa_var_samplerate * inp_alsa_var_frames_target;  // in micro seconds
 	if (snd_pcm_hw_params_set_period_time_near(priv->chandle, hwparams, &tmp, &dir) < 0){
 		visual_log(VISUAL_LOG_CRITICAL, _("Error setting period time"));
 		snd_pcm_hw_params_free(hwparams);
 		return(-1);
 	}
 
-	tmp = 1000000*4;
+	tmp = 1000 * 1000 / inp_alsa_var_samplerate * inp_alsa_var_frames_target;  // in micro seconds
```

Comment:
This suffers from the integer division problem above.

Reply:
Addressed above at https://github.com/Libvisual/libvisual/pull/154#discussion_r1070397953